docs/Build.md

```diff
@@ -0,0 +1,145 @@
+
+### Treeshaking
+
+To maintain a small built bundle size, we take advantage of [Webpack's treeshaking capabilities](https://webpack.js.org/guides/tree-shaking/) to only include what is used in the built output. This will require developers of the UI to be aware of the following two points:
+
+1. Where possible, use ECMAscript 6/ES2015 versions of 3rd party dependencies. For example, rather than using `lodash`, use `lodash-es`. This is so the Webpack can detect the various `import` and `export` of modules, and thus remove any which are never used.
+2. Modules exported in this codebase are expected to be pure - ie no side effects while `import`ing or `export`ing the content of a module. As a result, `sideEffects: false` is set in the `package.json`, meaning Webpack when building in production mode can prune functions and whole modules if required (and thus side effects, if any, may not occur if that module is not required). If a module/file does have side effects, these can be included still, by adding them to `sideEffects` as an array, i.e. `sideEffects: ['path/to/module/with/sideEffect.js']`. This should however only be a last resort.
+
+### Webpack setup
+
+Webpack (and it's plugins) are highly customisable. This section will detail the choices we have made around how and what is built, where it is built to, and the roles and responsibilities of the various plugins which enable this.
+
+#### Webpack aliases
+
+The UI makes extensive use of Webpack's [alias](https://webpack.js.org/configuration/resolve/#resolvealias) features. This is done for two reasons:
+
+1. It allows for abstraction at build time. By importing view layers via an alias the resolution of that alias can be changed at build time, enabling the ability to swap view layer implementations without needing to make changes in the production code.
+2. It makes importing code far cleaner and easier. For example, a traditional import of a module in another directory:
+
+```
+import { myFunction } from '../../Modules/MyModule';
+```
+
+with the following Webpack configuration:
+
+```
+...
+alias: {
+    MyModule: path.resolve(__dirname, 'Modules/MyModule')
+}
+...
+```
+
+becomes
+
+```
+import { myFunction } from 'MyModule'
+```
+
+For convenience, all top level module directories will be automatically aliased via a helper in the webpack configuration, as well defining a view alias.
+
+#### Webpack configuration and plugins
+
+In addition to aliasing, the webpack configuration will be as follows:
+
+| Option        | Value           | Purpose  |
+| ------------- | ------------- | -----:|
+| entry      | `client/Bootstrap/index.js` | Build entry point. Note the `scss/css` is expected to be imported by individual view layer code, which is imported (directly or indirectly) from this file  |
```

Comment:
Casual mention over scss/css imports - is that (to be) mentioned elsewhere?

Reply:
I was going to be explicit in a 'Code Style' doc PR, but my suggestion would be a (s)css import per component (as a part of a given view layer). The CSS optimisation plugin will then deduplicate down to a single set of class names. This should keep builds fast when developing (ie just one file changing) but also minimal when built in production. I will update the doc to mention details to follow - plus a few other small tweaks.

Reply:
Will optimisation namespace where duplicates happen? Or fail build (hopefully this)?

Reply:
OOTB, it will merge:
File one:

```
.foo {
color: red;
height: 100%;
}
```
File two:
```
.foo {
color: green;
}
bar: {
color: red;
}
```

will become:

```
.foo {
color: green;
height: 100%;
}
bar: {
color: red;
}
```

when built, with standard css precedence (ie later takes precedence) based on import order.

If it is ok, I will look at options here, and if we want to fail on conflicts - thoughts welcome. I can see pro's and cons either way. Based on discussion, I will update the documentation here with our decision in a follow up PR. Worth noting this optimisation only applies on production builds - when in development, we would see multiple (identical) class names (say), and where they originate from.

Reply:
Yep sounds good. My main concern is over whether we have to enforce name-spacing as a code style, or it can be explicitly forced.